### DISCUSSION TITLE
Clouseau &skip parameter no more working

### DISCUSSION THREAD

**Discussion by vintzl** [apache/couchdb#3313] [Q&A] [answered]
## Description

Cloudant &skip parameter no more works

## Steps to Reproduce

Try to follow the guide here:
- https://docs.couchdb.org/en/latest/api/ddoc/views.html#using-limits-and-skipping-rows

with this:
`GET /recipes/_design/recipes/_view/by_title?limit=3&skip=2 HTTP/1.1`

the limit is respected, but `&skip` parameter is ignored

## Expected Behaviour

Rows are skipped as before.

## Your Environment

* CouchDB version used: `3.1.1~buster`

* Browser name and version: `curl 7.64.0-4+deb10u1`
* Operating system and version: `Linux 5.9.0-5-amd64 #1 SMP Debian 5.9.15-1 (2020-12-17) x86_64 GNU/Linux`

## Additional Context

clouseau-2.17.0

**Comment by wohali** [comment]
I cannot confirm this. Worked example:

```javascript
$ curl -X PUT localhost:15984/abcd
{"ok":true}
$ curl -X PUT localhost:15984/abcd/a -d '{"name":"a"}'
{"ok":true,"id":"a","rev":"1-5be3091b08e7e742e372901aede1796a"}
$ curl -X PUT localhost:15984/abcd/b -d '{"name":"b"}'
{"ok":true,"id":"b","rev":"1-f8530e40c182151cdbc43993a087fa21"}
$ curl -X PUT localhost:15984/abcd/c -d '{"name":"c"}'
{"ok":true,"id":"c","rev":"1-03a8cbc32975d2f58295a1ea2617d965"}
$ curl -X PUT localhost:15984/abcd/d -d '{"name":"d"}'
{"ok":true,"id":"d","rev":"1-de10dc97e9b77ce7a64a7df692d7f845"}
$ curl -X PUT localhost:15984/abcd/e -d '{"name":"e"}'
{"ok":true,"id":"e","rev":"1-3355d6ee1cad28fa5ef78da20a2cb196"}
$
$
$ curl -X PUT localhost:15984/abcd/_design/view -d '{"views":{"by_name":{"map":"function(doc){emit(doc.name, null);}"}}}'
{"ok":true,"id":"_design/view","rev":"1-7b659b3fdfb061ee432189c496b94932"}
$
$
$ curl localhost:15984/abcd/_design/view/_view/by_name
{"total_rows":5,"offset":0,"rows":[
{"id":"a","key":"a","value":null},
{"id":"b","key":"b","value":null},
{"id":"c","key":"c","value":null},
{"id":"d","key":"d","value":null},
{"id":"e","key":"e","value":null}
]}
$
$
$ curl localhost:15984/abcd/_design/view/_view/by_name?limit=3
{"total_rows":5,"offset":0,"rows":[
{"id":"a","key":"a","value":null},
{"id":"b","key":"b","value":null},
{"id":"c","key":"c","value":null}
]}
$
$
$ curl "localhost:15984/abcd/_design/view/_view/by_name?limit=3&skip=2"
{"total_rows":5,"offset":2,"rows":[
{"id":"c","key":"c","value":null},
{"id":"d","key":"d","value":null},
{"id":"e","key":"e","value":null}
]}
```

Be sure to put the url in "quotation marks" if you are using a `&` to prevent the shell from interpreting this as a request to run `curl` as a background job,

**Reply by vintzl** [reply]
The URL is in quotation marks, and I can reproduce this issue in browsers, like Firefox, etc…

It used to work when I was using lucene from here
- https://github.com/rnewson/couchdb-lucene

Maybe I am doing something wrong or I am missing something, but it does not work.

**Reply by wohali** [reply]
Is this specific to `clouseau`? Can you supply your specific view code?

if it's Lucene-specific you may need @rnewson 's help, and I think he's out until next week.

**Reply by vintzl** [reply]
Yes it is specific to clouseau.
I have no view, only a design document for the index.

**Comment by kocolosk** [comment]
Should this description reference "Clouseau" instead of "Cloudant"?

**Comment by rnewson** [comment, marked as answer]
couchdb-lucene is not clouseau, the former supports `skip` but the latter does not.

**Reply by vintzl** [reply]
Wasn't [Apache Lucene](https://docs.couchdb.org/en/latest/install/search.html) supposed to be the official integration of [this repo](
    https://github.com/rnewson/couchdb-lucene
) inside CouchDB?

How can I do `skip` now?

**Reply by wohali** [reply]
No, that repo is only for CouchDB 1.x. The clouseau service is required for CouchDB 2.x/3.x.

As for `skip`, [now you use `bookmarks`](https://docs.couchdb.org/en/latest/ddocs/search.html#ddoc-search):

> By default, a search index response returns 25 rows. The number of rows that is returned can be changed by using the limit parameter. Each response includes a bookmark field. You can include the value of the bookmark field in later queries to look through the responses.
